sdk/core/core/inc/az_http_policy.h

```diff
@@ -58,6 +59,16 @@ AZ_NODISCARD az_result az_http_pipeline_policy_authentication(
     az_http_request_builder * const hrb,
     az_mut_span const * const response);
 
+AZ_INLINE AZ_NODISCARD az_result
+az_http_pipeline_policy_authentication_with_secret_credential_build(
```

Comment:
with_ `client_`credential

Reply:
az_http_pipeline_policy_build_auth_client_secret ?

Reply:
I think maybe we don't need a specialized version for client_secret_credential - just accept `void* credential`, because there will be more and more auth methods, presumably some of them will be module-specific.

Reply:
Also, it is probably `_init`, not `_build`